### PR TITLE
fixes #10281 - do not display inaccessible red hat repos for enable

### PR DESCRIPTION
**Pull request by thomasmckay**
Orphaned repos that have been enabled and sync'ed are the only ones shown.

When a subscription is removed (via expiration or manifest refresh), the cdn is no longer available for the repos that came from that sub. This means that new listings are not available when expanding in the Red Hat Repositories page. Previously synced repos should remain visible on the page. This PR fixes the UI page for this case.

### CODE REVIEW

File: app/models/katello/product.rb [hunk 1]
```diff
@@ -205,9 +205,14 @@ def to_action_input
...
-                :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
-      ::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
+      product_certificate = certificate
+      if product_certificate.nil?
+        nil
+      else
+        certs = { :ssl_client_cert => OpenSSL::X509::Certificate.new(product_certificate),
+                  :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
+        ::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
+      end
```
Comment (ehelms) [inline app/models/katello/product.rb]: This might be a little cleaner:

```
return unless (product_certificate = certificate)
certs = { :ssl_client_cert => OpenSSL::X509::Certificate.new(product_certificate),
                :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
```

File: app/lib/actions/katello/repository_set/scan_cdn.rb [hunk 2]
```diff
@@ -32,14 +32,17 @@ def fetch_results
...
+            substitutor.substitute_vars(content.contentUrl).map do |(substitutions, path)|
               prepare_result(substitutions, path)
             end
           end
         end
 
         def cdn_var_substitutor
-          product.cdn_resource.substitutor
+          cdn_resource = product.cdn_resource
+          cdn_resource.nil? ? nil : cdn_resource.substitutor
```
Comment (ehelms) [inline app/lib/actions/katello/repository_set/scan_cdn.rb]:
```
return unless (cdn_resource = product.cdn_resource)
cdn_resource.substitutor
```

File: app/controllers/katello/products_controller.rb [hunk 3]
```diff
@@ -30,6 +30,27 @@ def available_repositories
...
+                  :content_id => product_repo.content_id,
+                  :substitutions => {
+                      :basearch => product_repo.arch,
+                      :releasever => product_repo.minor
+                  },
+                  :enabled => true,
+                  :promoted => product_repo.promoted?,
+                  :registry_name => product_repo.docker_upstream_name
+              }
+            end
```
Comment (ehelms) [inline app/controllers/katello/products_controller.rb]: This logic is a little hard to follow with its intent. Breaking the logic chunks into methods with names that convey intent might help.

**Comment by ehelms** [comment]
I read through the issue, but I am still a little unclear what inaccessible repositories are defined as. Can you dumb it down for me?

**Comment by thomasmckay** [comment]
@ehelms added some description to this PR and made suggested changes. Thanks!

File: app/controllers/katello/products_controller.rb [hunk 4]
```diff
@@ -95,5 +89,39 @@ def substitutions
     def task_error(task)
       task.failed_steps.first.action(task.execution_plan).steps.map { |s| s.try(:error).try(:message) }.reject(&:blank?).join(', ')
     end
+
+    def available_kickstart_repos(repos)
```
Comment (jlsherrill) [inline app/controllers/katello/products_controller.rb]: I would rename this to something like  'exclude_rolling_kickstart_repos'.   Not the greatest name, but i was REALLY confused by the name 'available_kickstart_repos'

Comment (thomasmckay) [inline app/controllers/katello/products_controller.rb]: done

**Comment by jlsherrill** [comment]
ACK from me

**Comment by ehelms** [comment]
Please to squash commits.
